src/Data/Form.purs

```diff
@@ -127,6 +138,126 @@ instance decodeInputSource :: (DecodeJson a) => DecodeJson (InputSource a) where
 instance arbitraryInputSource :: (Arbitrary a) => Arbitrary (InputSource a) where
   arbitrary = genericArbitrary
 
+userInput :: forall a. InputSource a -> Maybe a
+userInput = case _ of
+  UserInput x -> Just x
+  Invalid _ -> Nothing
+  NotSet -> Nothing
```

Comment:
Hmm, now that we have this helper we can probably simplify my `getValue` helper to: 
```purescript
getValue { value, default } = userInput value <|> defualt
```

Reply:
Might also be able to shorten this to
```purescript
userInput = case _ of
  UserInput x -> Just x
  otherwise -> Nothing
```
Since judging from the function name everything but `UserInput x` should fail.

Reply:
The [bug][] you pointed out below is the main reason I try to write all the cases explicitly when I remember. I didn't in that case, and luckily you caught it. But, this is work the compiler should be doing for us. I'm of two minds with this: I don't enjoy writing the cases explicitly, but I really dislike having to remember to check cases. If you're alright with it, can we keep this one explicit so we're less likely to run into another one of these [bugs][bug]?

[bug]: https://github.com/citizennet/purescript-lynx/pull/8#discussion_r263152203

Reply:
I'm fine with leaving it. I did figure that was your reason, and I totally agree with it. I was thinking we could make an exception here since this is a case where we're only looking to match one condition, and we know any other condition should fail. But I also think it's valuable to establish the practice of explicitly covering each case when possible, so we can avoid bugs like below.

Reply:
Yeah, that seems fair. Let's wildcard this up!.